### PR TITLE
Request permissions for Android push notifications and refactor

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
Ensure android permissions for push notifications are requested.

On top of it, refactor native code to use the Networking library for consistency, simplification and to reduce the amount of times the token is retrieved from the KeyStore.

Also the user profile images now use an in-memory cache to reduce the amount of requests. (iOS uses a disk cache instead)

#### Release Note
```release-note
Ensures the permissions required to receive push notifications is always requested if needed.
```

### CODE REVIEW

File: android/app/src/main/java/com/mattermost/rnbeta/NotificationReplyBroadcastReceiver.java [hunk 1]
```diff
@@ -53,12 +41,7 @@ public void onReceive(Context context, Intent intent) {
...
             if (serverUrl != null) {
-                final ReactApplicationContext reactApplicationContext = new ReactApplicationContext(context);
-                final String token = Credentials.getCredentialsForServerSync(reactApplicationContext, serverUrl);
-
-                if (token != null) {
-                    replyToMessage(serverUrl, token, notificationId, message);
-                }
+                    replyToMessage(serverUrl, notificationId, message);
             } else {
                 onReplyFailed(notificationId);
```
Comment (avinashlng1080) [inline android/app/src/main/java/com/mattermost/rnbeta/NotificationReplyBroadcastReceiver.java:46]: I'm not 100% here but is it creating an infinite loop?

**Review by avinashlng1080 — APPROVED** [review]

File: android/app/src/main/java/com/mattermost/rnbeta/NotificationReplyBroadcastReceiver.java (hunk 1 above)
Comment (enahum) [reply android/app/src/main/java/com/mattermost/rnbeta/NotificationReplyBroadcastReceiver.java:46]: How?

Comment (avinashlng1080) [reply android/app/src/main/java/com/mattermost/rnbeta/NotificationReplyBroadcastReceiver.java:46]: At line 46, `onReplyFailed` will invoke `recreateNotification`  which in turn will call `notificationManager.notify(notificationId, notification);` .  Now, if the `serverUrl ` was never in the `bundle`.  would this trigger a loop and always hit the `if block` at line 43?

Comment (enahum) [reply android/app/src/main/java/com/mattermost/rnbeta/NotificationReplyBroadcastReceiver.java:46]: no cause `onReceive` is not being called after notify.. there are two different onReceive, this one which is when you reply directly in the notification and the `onReceive` when a remote notification arrives.

Feel free to give it a try

Comment (avinashlng1080) [reply android/app/src/main/java/com/mattermost/rnbeta/NotificationReplyBroadcastReceiver.java:46]: Yes thanks :)

**Review by larkox — APPROVED** [review]

**Comment by enahum** [comment]
manually cherry-pick to 2.0

**Pull request merged**